ext/ffi/lib.rs

```diff
@@ -450,22 +473,30 @@ fn ffi_call(args: FfiCallArgs, symbol: &Symbol) -> Result<Value, AnyError> {
     .map(|buffer| buffer.as_ref().map(|buffer| &buffer[..]))
     .collect();
 
-  let native_values = symbol
+  let mut native_values: Vec<NativeValue> = vec![];
+
+  for (&native_type, value) in symbol
     .parameter_types
     .iter()
     .zip(args.parameters.into_iter())
-    .map(|(&native_type, value)| {
-      if let NativeType::Pointer = native_type {
-        if let Some(idx) = value.as_u64() {
-          if let Some(&Some(buf)) = buffers.get(idx as usize) {
-            return NativeValue::buffer(buf.as_ptr());
-          }
+  {
+    match native_type {
+      NativeType::Pointer => match value.as_u64() {
+        Some(idx) => {
+          let buf = buffers.get(idx as usize).unwrap().unwrap();
```

Comment:
These unwraps should be thrown as errors too.

Reply:
I thought it'd not be required since these are actually `Option` and always expected to be there according to the JS side which calls it. I'll throw these anyway, but what should be the error message?

Reply:
Actually I agree, these can be only reachable if our internal JS code is buggy. Let's keep them as panics we would want users to report them as bugs.